### PR TITLE
Update DynamoDb to support creating tables and dev envs

### PR DESCRIPTION
**Pull request by TinyPandas**
# Description
DynamoDb related changes
- Support creating tables outside of AWS Console.
- Tables referenced from dev environment will be prefixed with `dev_` to retain production stability.
- Move `logic` package to correct top-level packages.
  - `logic.data.managers` -> `managers`
  - `logic.data.models` -> `beans`
  - Constants -> `utils`

# How Has This Been Tested?

No unit tests.
Actual tests against the new code was done and verified tables would not clash.

# Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my code
- [ ] I have commented my code appropriately, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by Ron4242 — APPROVED** [review]
LGTM!

File: src/main/java/org/rsa/Bot.java [hunk 1]
```diff
@@ -28,7 +27,7 @@
 public class Bot {
 
     private static final String VERSION = "v1.2.7";
-    private static boolean isDev = false;
+    public static boolean isDev = true;
```
[the remark this answers is not in the document]
Comment (TinyPandas) [reply src/main/java/org/rsa/Bot.java:30]: truuueeee, should probably find a dynamic way to set this. :/

File: src/main/java/org/rsa/aws/client/DynamoClient.java
[the remark this answers is not in the document]
Comment (TinyPandas) [reply src/main/java/org/rsa/aws/client/DynamoClient.java:1]: I disagree. Naming of `Singleton` doesn't exist in industry and makes no sense here, imo.

File: src/main/java/org/rsa/Bot.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (TinyPandas) [reply src/main/java/org/rsa/Bot.java:35]:
```
    private static String getBotToken() {
        String botToken = System.getenv("BOT_TOKEN");

        if (null == botToken) {
            isDev = false;
            botToken = getValue(SecretsManager.BOT_TOKEN_KEY);
        }

        return botToken;
    }
```

This logic is already being treated appropriately in the `getBotToken` method. As the bot token is not stored in system env on prod, and is in dev environments (or should be).

Should probably have a fallback just in-case someone shortcuts the bot token logic.. 🙄

**Pull request merged**